src/zkevm_specs/evm/execution/create.py

```diff
@@ -0,0 +1,149 @@
+    offset = instruction.stack_pop()
+    size = instruction.stack_pop()
+    contract_address = instruction.stack_push()
+
+    depth = instruction.call_context_lookup(CallContextFieldTag.Depth)
+    tx_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
+    caller_address = instruction.call_context_lookup(CallContextFieldTag.CallerAddress)
+    nonce, nonce_prev = instruction.account_write(caller_address, AccountFieldTag.Nonce)
+    is_success = instruction.call_context_lookup(CallContextFieldTag.IsSuccess)
+    is_static = instruction.call_context_lookup(CallContextFieldTag.IsStatic)
+    reversion_info = instruction.reversion_info()
+    contract_address = instruction.generate_contract_address(caller_address, nonce_prev)
+
+    # Verify depth is less than 1024
+    instruction.range_lookup(depth, CALL_CREATE_DEPTH)
+
+    # ErrNonceUintOverflow constraint
+    (is_not_overflow, _) = instruction.compare(nonce, nonce_prev, N_BYTES_U64)
+    instruction.is_zero(is_not_overflow)
+
+    # add contract address to access list
+    instruction.add_account_to_access_list(tx_id, contract_address)
+
+    # ErrContractAddressCollision constraint
+    code_hash = instruction.account_read(contract_address, AccountFieldTag.CodeHash)
+    instruction.constrain_equal(code_hash, RLC(EMPTY_HASH))
+
+    # Propagate is_persistent
+    callee_reversion_info = instruction.reversion_info(call_id=callee_call_id)
+    instruction.constrain_equal(
+        callee_reversion_info.is_persistent,
+        reversion_info.is_persistent * is_success,
+    )
+
+    # can't be a STATICCALL
+    instruction.is_zero(is_static)
+
+    # transfer value from caller to contract address
+    instruction.transfer(caller_address, contract_address, value)
+
+    # gas cost of memory expansion
+    (next_memory_size, memory_expansion_gas_cost,) = instruction.memory_expansion_constant_length(
+        offset,
+        size,
+    )
+
+    # gas cost of CREATE = GAS_COST_CREATE + memory expansion + GAS_COST_CODE_DEPOSIT * len(byte_code)
+    # the last one (GAS_COST_CODE_DEPOSIT * len(byte_code)) would be calculated in `return_revert`
+    gas_left = instruction.curr.gas_left
+    gas_cost = GAS_COST_CREATE + memory_expansion_gas_cost
+    gas_available = gas_left - gas_cost
+
+    # Apply EIP 150
+    one_64th_gas, _ = instruction.constant_divmod(gas_available, FQ(64), N_BYTES_GAS)
+    all_but_one_64th_gas = gas_available - one_64th_gas
+    is_u64_gas = instruction.is_zero(
+        instruction.sum(instruction.rlc_encode(gas_left).le_bytes[N_BYTES_GAS:])
+    )
+    callee_gas_left = instruction.select(
+        is_u64_gas,
+        instruction.min(all_but_one_64th_gas, gas_left, N_BYTES_GAS),
+        all_but_one_64th_gas,
+    )
+
+    # calculate code_hash = hash(init_code)
+    copy_rwc_inc, rlc_acc = instruction.copy_lookup(
+        instruction.curr.call_id,  # src_id
+        CopyDataTypeTag.Memory,  # src_type
+        code_hash,  # dst_id
```

Comment:
Then we use the updated `code_hash` here instead of the old one.

Reply:
fixed in [98f9722](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/355/commits/98f972291b6884c84227ca761e668e5999c2c525)